### PR TITLE
bugfix for update with empty buffer; remove duplicate variable _weight_sum in PrioritizedReplayBuffer

### PR DESCRIPTION
**Pull request by youkaichao**
- [ ] I have marked all applicable categories:
    + [ ] exception-raising fix
    + [ ] algorithm implementation fix
    + [ ] documentation modification
    + [ ] new feature
- [ ] If applicable, I have mentioned the relevant/related issue(s)

Less important but also useful:

- [ ] I have visited the [source website], and in particular read the [known issues]
- [ ] I have searched through the [issue tracker] and [issue categories] for duplicates
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, torch, sys
  print(tianshou.__version__, torch.__version__, sys.version, sys.platform)
  ```

  [source website]: https://github.com/thu-ml/tianshou
  [known issues]: https://github.com/thu-ml/tianshou/#faq-and-known-issues
  [issue categories]: https://github.com/thu-ml/tianshou/projects/2
  [issue tracker]: https://github.com/thu-ml/tianshou/issues?q=

### CODE REVIEW

File: tianshou/data/buffer.py [hunk 1]
```diff
@@ -151,6 +151,8 @@ def _add_to_buffer(self, name: str, inst: Any) -> None:
 
     def update(self, buffer: 'ReplayBuffer') -> None:
         """Move the data from the given buffer to self."""
+        if len(buffer) == 0:
+            return
```
Comment (youkaichao) [inline tianshou/data/buffer.py:155]: bugfix for update with empty buffer

**Comment by youkaichao** [comment]
Currently, in the ``PrioritizedReplayBuffer.sample()``, there is ``self.weight / self.weight.sum()``, which means each time the code will calculate the sum of weight, so we do not need a ``self._weight_sum`` variable any more.

In addition, since it is a ``PrioritizedReplayBuffer``, it makes no sense to pass ``importance_sample`` argument to ``sample()`` (``importance_sample`` should always be ``True``), which also makes the interface ``sample()`` consistent with other replaybuffers.

File: tianshou/data/buffer.py (hunk 1 above)
Comment (duburcqa) [reply tianshou/data/buffer.py:155]: Why not raise an exception ? Is doing this allowed ?

Comment (youkaichao) [reply tianshou/data/buffer.py:155]: Update a buffer with another empty buffer is allowed. Just like extending a list with another list ``a += b``. We just need to assure it behaves correctly.

**Review by duburcqa — APPROVED** [review]

**Comment by Trinkle23897** [comment]
I’m wondering which one is better in prioritized buffer: segment tree or np.sum. Maybe I’ll do this tomorrow.

**Pull request merged**